python/nav/models/sql/changes/sc.04.10.0001.sql

```diff
@@ -0,0 +1 @@
+DELETE FROM alertsender WHERE handler='jabber';
```

Comment:
While this may seem fine on the surface, I'm afraid this may confuse anyone who has been actively using Jabber in their alert profile. Any subscription defined in an alertprofile, related to an alertaddress of type `jabber` will disappear entirely because this alertsender is being removed, and the user will be given no chance to change their subscriptions to dispatch alerts somewhere else.

A more graceful solution would keep the subscriptions, but leave them in some sort of invalid state, where they would be marked as such in the Alert Profiles UI, but otherwise ignored by the alert engine. The user would then be able to change the alert address of their invalid subscriptions. However, this solution would be a bit more involved. I'm open for a discussion on how to solve it.

Reply:
While I agree that this might technically be better, I have no idea how to test this.

Reply:
No idea how to test it, or how to write an automated test for it?

I think maybe the simplest solution SQL-wise, would be to change the constraints for `AlertAddress`, so that when an `AlertSender` is deleted, the AlertSender foreign key can be nulled out (instead of cascading the delete, such as now).

Then, modifications are needed to the code to account for AlertAddress entries that have no AlertSender reference. I suppose:

1. AlertEngine should do nothing if a subscription matches and the address is invalid (Except log a warning)
2. AlertProfiles should heavily draw attention to an invalid address anywhere it appears in the UI (some kind of red mark, with a warning to explain that the address is invalid because the alert dispatch mechanism has been removed from NAV).